[Writing Tools] Split up HTMLConverter into different files for each type of converter

rr-codes opened the pull request:
#### dbae949044fcf7de6a8ce85a63149303301f1747
<pre>
[Writing Tools] Split up HTMLConverter into different files for each type of converter
<a href="https://bugs.webkit.org/show_bug.cgi?id=296031">https://bugs.webkit.org/show_bug.cgi?id=296031</a>
<a href="https://rdar.apple.com/155935345">rdar://155935345</a>

Reviewed by Aditya Keerthi.

The HTMLConverter.mm file has two implementations of HTML conversion, both of which are relatively complicated.
To improve readability and maintainability, split these implementations into their own files.

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/editing/cocoa/EditingHTMLConverter.mm: Added.
(WebCore::preferredFilenameForElement):
(WebCore::fileWrapperForElement):
(WebCore::attributedStringWithAttachmentForFileWrapper):
(WebCore::attributedStringWithAttachmentForElement):
(WebCore::elementQualifiesForWritingToolsPreservation):
(WebCore::hasAncestorQualifyingForWritingToolsPreservation):
(WebCore::enclosingLinkElement):
(WebCore::updateAttributes):
(WebCore::editingAttributedStringInternal):
(WebCore::editingAttributedString):
(WebCore::editingAttributedStringReplacingNoBreakSpace):
* Source/WebCore/editing/cocoa/HTMLConverter.h:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(WebCore::WebDefaultFont):
(WebCore::fileWrapperForURL):
(HTMLConverterCaches::isAncestorsOfStartToBeConverted const): Deleted.
(): Deleted.
(HTMLConverter::HTMLConverter): Deleted.
(HTMLConverter::convert): Deleted.
(WebDefaultFont): Deleted.
(_fontForNameAndSize): Deleted.
(defaultParagraphStyle): Deleted.
(HTMLConverterCaches::computedStylePropertyForElement): Deleted.
(HTMLConverterCaches::inlineStylePropertyForElement): Deleted.
(stringFromCSSValue): Deleted.
(HTMLConverterCaches::propertyValueForNode): Deleted.
(floatValueFromPrimitiveValue): Deleted.
(HTMLConverterCaches::floatPropertyValueForNode): Deleted.
(_shadowForShadowStyle): Deleted.
(HTMLConverterCaches::isBlockElement): Deleted.
(HTMLConverterCaches::elementHasOwnBackgroundColor): Deleted.
(HTMLConverter::_blockLevelElementForNode): Deleted.
(normalizedColor): Deleted.
(HTMLConverterCaches::colorPropertyValueForNode): Deleted.
(HTMLConverter::_colorForElement): Deleted.
(_font): Deleted.
(HTMLConverter::computedAttributesForElement): Deleted.
(HTMLConverter::attributesForElement): Deleted.
(HTMLConverter::aggregatedAttributesForAncestors): Deleted.
(HTMLConverter::aggregatedAttributesForElementAndItsAncestors): Deleted.
(HTMLConverter::_newParagraphForElement): Deleted.
(HTMLConverter::_newLineForElement): Deleted.
(HTMLConverter::_newTabForElement): Deleted.
(_WebMessageDocumentClass): Deleted.
(HTMLConverter::_addMultiRepresentationHEICAttachmentForImageElement): Deleted.
(HTMLConverter::_addAttachmentForElement): Deleted.
(HTMLConverter::_addQuoteForElement): Deleted.
(HTMLConverter::_addValue): Deleted.
(HTMLConverter::_fillInBlock): Deleted.
(read2DigitNumber): Deleted.
(_dateForString): Deleted.
(_colCompare): Deleted.
(HTMLConverter::_processMetaElementWithName): Deleted.
(HTMLConverter::_processHeadElement): Deleted.
(HTMLConverter::_enterBlockquote): Deleted.
(HTMLConverter::_exitBlockquote): Deleted.
(HTMLConverter::_enterElement): Deleted.
(HTMLConverter::_addLinkForElement): Deleted.
(HTMLConverter::_addTableForElement): Deleted.
(HTMLConverter::_addTableCellForElement): Deleted.
(HTMLConverter::_processElement): Deleted.
(HTMLConverter::_addMarkersToList): Deleted.
(HTMLConverter::_exitElement): Deleted.
(HTMLConverter::_processText): Deleted.
(HTMLConverter::_traverseNode): Deleted.
(HTMLConverter::_traverseFooterNode): Deleted.
(HTMLConverter::_adjustTrailingNewline): Deleted.
(HTMLConverterCaches::cacheAncestorsOfStartToBeConverted): Deleted.
(fileWrapperForURL): Deleted.
(preferredFilenameForElement): Deleted.
(fileWrapperForElement): Deleted.
(attributedStringWithAttachmentForFileWrapper): Deleted.
(attributedStringWithAttachmentForElement): Deleted.
(elementQualifiesForWritingToolsPreservation): Deleted.
(hasAncestorQualifyingForWritingToolsPreservation): Deleted.
(enclosingLinkElement): Deleted.
(updateAttributes): Deleted.
(WebCore::attributedString): Deleted.
(WebCore::editingAttributedStringInternal): Deleted.
(WebCore::editingAttributedString): Deleted.
(WebCore::editingAttributedStringReplacingNoBreakSpace): Deleted.
* Source/WebCore/editing/cocoa/NodeHTMLConverter.mm: Copied from Source/WebCore/editing/cocoa/HTMLConverter.mm.
(HTMLConverterCaches::isAncestorsOfStartToBeConverted const):
(HTMLConverter::HTMLConverter):
(HTMLConverter::convert):
(_fontForNameAndSize):
(defaultParagraphStyle):
(HTMLConverterCaches::computedStylePropertyForElement):
(HTMLConverterCaches::inlineStylePropertyForElement):
(stringFromCSSValue):
(HTMLConverterCaches::propertyValueForNode):
(floatValueFromPrimitiveValue):
(HTMLConverterCaches::floatPropertyValueForNode):
(_shadowForShadowStyle):
(HTMLConverterCaches::isBlockElement):
(HTMLConverterCaches::elementHasOwnBackgroundColor):
(HTMLConverter::_blockLevelElementForNode):
(normalizedColor):
(HTMLConverterCaches::colorPropertyValueForNode):
(HTMLConverter::_colorForElement):
(_font):
(HTMLConverter::computedAttributesForElement):
(HTMLConverter::attributesForElement):
(HTMLConverter::aggregatedAttributesForAncestors):
(HTMLConverter::aggregatedAttributesForElementAndItsAncestors):
(HTMLConverter::_newParagraphForElement):
(HTMLConverter::_newLineForElement):
(HTMLConverter::_newTabForElement):
(_WebMessageDocumentClass):
(HTMLConverter::_addMultiRepresentationHEICAttachmentForImageElement):
(HTMLConverter::_addAttachmentForElement):
(HTMLConverter::_addQuoteForElement):
(HTMLConverter::_addValue):
(HTMLConverter::_fillInBlock):
(read2DigitNumber):
(_dateForString):
(_colCompare):
(HTMLConverter::_processMetaElementWithName):
(HTMLConverter::_processHeadElement):
(HTMLConverter::_enterBlockquote):
(HTMLConverter::_exitBlockquote):
(HTMLConverter::_enterElement):
(HTMLConverter::_addLinkForElement):
(HTMLConverter::_addTableForElement):
(HTMLConverter::_addTableCellForElement):
(HTMLConverter::_processElement):
(HTMLConverter::_addMarkersToList):
(HTMLConverter::_exitElement):
(HTMLConverter::_processText):
(HTMLConverter::_traverseNode):
(HTMLConverter::_traverseFooterNode):
(HTMLConverter::_adjustTrailingNewline):
(HTMLConverterCaches::cacheAncestorsOfStartToBeConverted):
(WebCore::attributedString):

Canonical link: <a href="https://commits.webkit.org/297624@main">https://commits.webkit.org/297624@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c4b466f7fe5a23d2cf1163bf9297a463ac7af93

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112341 "32 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32073 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22550 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118420 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62709 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114303 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32725 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40636 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85339 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/36027 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115288 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26112 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101065 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65770 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25410 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19201 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62265 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/95496 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121745 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39415 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/29336 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/94147 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39796 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97306 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93973 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24029 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39212 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17006 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35465 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39303 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44791 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38938 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42275 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40681 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/SourcesCocoa.txt:
```diff
@@ -197,6 +197,8 @@ editing/cocoa/FontAttributeChangesCocoa.mm
 editing/cocoa/FontShadowCocoa.mm
 editing/cocoa/HTMLConverter.mm @no-unify
+editing/cocoa/NodeHTMLConverter.mm @no-unify
+editing/cocoa/EditingHTMLConverter.mm @no-unify
```
Why @no-unify? 

Honestly, I am not clear on what the criteria for anything being @no-unify is, so I am quite curious.

rr-codes replied:
neither am I 😅 but HTMLConverter.mm had it so I followed precedent; plus, doing so gives syntax highlighting and code completion when using Xcode 🤷

weinig replied:
I am not sure precedent without rationale is a good reason here. 

Having "syntax highlighting and code completion when using Xcode" working only in some files seems so lame. We really got either fix or remove the unified build :\.

rr-codes replied:
Ya, I agree with that. In the meantime though I think I'll leave things as-is since this isn't really changing anything from the status quo. (and that can be done in a separate change if desired)

weinig reviewed (commented):
It's pretty non-standard to have .cpp files without a correspondingly named .h file in WebCore. 

I think this would be more clear if `attributedString()` were in a `NodeHTMLConverter.h` and `editingAttributedString()`/`editingAttributedStringReplacingNoBreakSpace()` were in an `EditingHTMLConverter.h` and then anything shared between them could be in `HTMLConverter.h`

rr-codes committed:
[Writing Tools] Split up HTMLConverter into different files for each type of converter
https://bugs.webkit.org/show_bug.cgi?id=296031
rdar://155935345

Reviewed by Aditya Keerthi.

The HTMLConverter.mm file has two implementations of HTML conversion, both of which are relatively complicated.
To improve readability and maintainability, split these implementations into their own files.

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/editing/cocoa/EditingHTMLConverter.mm: Added.
(WebCore::preferredFilenameForElement):
(WebCore::fileWrapperForElement):
(WebCore::attributedStringWithAttachmentForFileWrapper):
(WebCore::attributedStringWithAttachmentForElement):
(WebCore::elementQualifiesForWritingToolsPreservation):
(WebCore::hasAncestorQualifyingForWritingToolsPreservation):
(WebCore::enclosingLinkElement):
(WebCore::updateAttributes):
(WebCore::editingAttributedStringInternal):
(WebCore::editingAttributedString):
(WebCore::editingAttributedStringReplacingNoBreakSpace):
* Source/WebCore/editing/cocoa/HTMLConverter.h:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(WebCore::WebDefaultFont):
(WebCore::fileWrapperForURL):
(HTMLConverterCaches::isAncestorsOfStartToBeConverted const): Deleted.
(): Deleted.
(HTMLConverter::HTMLConverter): Deleted.
(HTMLConverter::convert): Deleted.
(WebDefaultFont): Deleted.
(_fontForNameAndSize): Deleted.
(defaultParagraphStyle): Deleted.
(HTMLConverterCaches::computedStylePropertyForElement): Deleted.
(HTMLConverterCaches::inlineStylePropertyForElement): Deleted.
(stringFromCSSValue): Deleted.
(HTMLConverterCaches::propertyValueForNode): Deleted.
(floatValueFromPrimitiveValue): Deleted.
(HTMLConverterCaches::floatPropertyValueForNode): Deleted.
(_shadowForShadowStyle): Deleted.
(HTMLConverterCaches::isBlockElement): Deleted.
(HTMLConverterCaches::elementHasOwnBackgroundColor): Deleted.
(HTMLConverter::_blockLevelElementForNode): Deleted.
(normalizedColor): Deleted.
(HTMLConverterCaches::colorPropertyValueForNode): Deleted.
(HTMLConverter::_colorForElement): Deleted.
(_font): Deleted.
(HTMLConverter::computedAttributesForElement): Deleted.
(HTMLConverter::attributesForElement): Deleted.
(HTMLConverter::aggregatedAttributesForAncestors): Deleted.
(HTMLConverter::aggregatedAttributesForElementAndItsAncestors): Deleted.
(HTMLConverter::_newParagraphForElement): Deleted.
(HTMLConverter::_newLineForElement): Deleted.
(HTMLConverter::_newTabForElement): Deleted.
(_WebMessageDocumentClass): Deleted.
(HTMLConverter::_addMultiRepresentationHEICAttachmentForImageElement): Deleted.
(HTMLConverter::_addAttachmentForElement): Deleted.
(HTMLConverter::_addQuoteForElement): Deleted.
(HTMLConverter::_addValue): Deleted.
(HTMLConverter::_fillInBlock): Deleted.
(read2DigitNumber): Deleted.
(_dateForString): Deleted.
(_colCompare): Deleted.
(HTMLConverter::_processMetaElementWithName): Deleted.
(HTMLConverter::_processHeadElement): Deleted.
(HTMLConverter::_enterBlockquote): Deleted.
(HTMLConverter::_exitBlockquote): Deleted.
(HTMLConverter::_enterElement): Deleted.
(HTMLConverter::_addLinkForElement): Deleted.
(HTMLConverter::_addTableForElement): Deleted.
(HTMLConverter::_addTableCellForElement): Deleted.
(HTMLConverter::_processElement): Deleted.
(HTMLConverter::_addMarkersToList): Deleted.
(HTMLConverter::_exitElement): Deleted.
(HTMLConverter::_processText): Deleted.
(HTMLConverter::_traverseNode): Deleted.
(HTMLConverter::_traverseFooterNode): Deleted.
(HTMLConverter::_adjustTrailingNewline): Deleted.
(HTMLConverterCaches::cacheAncestorsOfStartToBeConverted): Deleted.
(fileWrapperForURL): Deleted.
(preferredFilenameForElement): Deleted.
(fileWrapperForElement): Deleted.
(attributedStringWithAttachmentForFileWrapper): Deleted.
(attributedStringWithAttachmentForElement): Deleted.
(elementQualifiesForWritingToolsPreservation): Deleted.
(hasAncestorQualifyingForWritingToolsPreservation): Deleted.
(enclosingLinkElement): Deleted.
(updateAttributes): Deleted.
(WebCore::attributedString): Deleted.
(WebCore::editingAttributedStringInternal): Deleted.
(WebCore::editingAttributedString): Deleted.
(WebCore::editingAttributedStringReplacingNoBreakSpace): Deleted.
* Source/WebCore/editing/cocoa/NodeHTMLConverter.mm: Copied from Source/WebCore/editing/cocoa/HTMLConverter.mm.
(HTMLConverterCaches::isAncestorsOfStartToBeConverted const):
(HTMLConverter::HTMLConverter):
(HTMLConverter::convert):
(_fontForNameAndSize):
(defaultParagraphStyle):
(HTMLConverterCaches::computedStylePropertyForElement):
(HTMLConverterCaches::inlineStylePropertyForElement):
(stringFromCSSValue):
(HTMLConverterCaches::propertyValueForNode):
(floatValueFromPrimitiveValue):
(HTMLConverterCaches::floatPropertyValueForNode):
(_shadowForShadowStyle):
(HTMLConverterCaches::isBlockElement):
(HTMLConverterCaches::elementHasOwnBackgroundColor):
(HTMLConverter::_blockLevelElementForNode):
(normalizedColor):
(HTMLConverterCaches::colorPropertyValueForNode):
(HTMLConverter::_colorForElement):
(_font):
(HTMLConverter::computedAttributesForElement):
(HTMLConverter::attributesForElement):
(HTMLConverter::aggregatedAttributesForAncestors):
(HTMLConverter::aggregatedAttributesForElementAndItsAncestors):
(HTMLConverter::_newParagraphForElement):
(HTMLConverter::_newLineForElement):
(HTMLConverter::_newTabForElement):
(_WebMessageDocumentClass):
(HTMLConverter::_addMultiRepresentationHEICAttachmentForImageElement):
(HTMLConverter::_addAttachmentForElement):
(HTMLConverter::_addQuoteForElement):
(HTMLConverter::_addValue):
(HTMLConverter::_fillInBlock):
(read2DigitNumber):
(_dateForString):
(_colCompare):
(HTMLConverter::_processMetaElementWithName):
(HTMLConverter::_processHeadElement):
(HTMLConverter::_enterBlockquote):
(HTMLConverter::_exitBlockquote):
(HTMLConverter::_enterElement):
(HTMLConverter::_addLinkForElement):
(HTMLConverter::_addTableForElement):
(HTMLConverter::_addTableCellForElement):
(HTMLConverter::_processElement):
(HTMLConverter::_addMarkersToList):
(HTMLConverter::_exitElement):
(HTMLConverter::_processText):
(HTMLConverter::_traverseNode):
(HTMLConverter::_traverseFooterNode):
(HTMLConverter::_adjustTrailingNewline):
(HTMLConverterCaches::cacheAncestorsOfStartToBeConverted):
(WebCore::attributedString):

Canonical link: https://commits.webkit.org/297624@main
